src/libsyntax/ast.rs

```diff
@@ -439,6 +439,22 @@ pub struct Crate {
     pub exported_macros: Vec<MacroDef>,
 }
 
+/// A spanned compile-time attribute list item.
+pub type MetaListItem = Spanned<MetaListItemKind>;
```

Comment:
A `MetaItem` can already be nested in itself, but it cannot appear in a list. Thus, the ability to nest isn't the differentiation, it's the fact that the type can appear inside of a list. If others feels strongly about this, I'm okay with changing the name.